### PR TITLE
Adding script to estimate the cost of a simulation

### PR DESCRIPTION
**Pull request by AugustoPeres**
This pull request adds a script to estimate the cost of one simulation. It submits just one tasks, wait for it to finish, gets the total time, and computes its cost.

Program output is as follows:

```
I1012 17:18:27.370823 140578044125248 machines.py:63] Registering machine group configuration with the API.
I1012 17:18:28.359893 140578044125248 machines.py:67] Registered machine group:
I1012 17:18:28.359977 140578044125248 machines.py:68]  > ID: bd36b6a1-617f-45e9-ab5c-6ad4f32587b3
I1012 17:18:28.360085 140578044125248 machines.py:69]  > Name: api-bd36b6a1-617f-45e9-ab5c-6ad4f32587b3
I1012 17:18:28.360340 140578044125248 machines.py:114] Starting a machine group. This may take up to a few minutes.
I1012 17:18:28.360370 140578044125248 machines.py:116] Note that stopping this local process will not interrupt the creation of the machine group. Please wait...
I1012 17:18:53.193711 140578044125248 machines.py:124] Machine group successfully started in 0.41 mins.
I1012 17:18:53.193801 140578044125248 machines.py:211] Name: api-bd36b6a1-617f-45e9-ab5c-6ad4f32587b3
I1012 17:18:53.193840 140578044125248 machines.py:212] Machine type: c2-standard-16
I1012 17:18:53.193861 140578044125248 machines.py:213] Number of machines: 1
I1012 17:18:53.193880 140578044125248 machines.py:214] Spot: False
I1012 17:18:53.193897 140578044125248 machines.py:215] Disk size: 40 GB
I1012 17:18:53.726830 140578044125248 machines.py:216] Estimated cost per hour: 0.919 $/h
I1012 17:18:53.827354 140578044125248 methods.py:284] Task ID: 1697127533798800741
I1012 17:18:53.827453 140578044125248 methods.py:81] Creating input archive...
I1012 17:18:53.832679 140578044125248 methods.py:88] Input archive created.
I1012 17:18:53.832753 140578044125248 methods.py:89] Input archive size: 27.12 KiB
I1012 17:18:53.832781 140578044125248 methods.py:92] Uploading input...
I1012 17:18:54.297517 140578044125248 methods.py:104] Input archive uploaded.
I1012 17:18:54.297678 140578044125248 methods.py:296] Task request submitted.
I1012 17:20:04.983391 140578044125248 machines.py:189] Estimated cloud cost per hour: 0.919 $/h
I1012 17:20:05.028886 140578044125248 estimate_costs.py:80] Simulation duration: 44.684656 seconds
I1012 17:20:05.028962 140578044125248 estimate_costs.py:82] Estimated cost per task: 0.011406999684444443
I1012 17:20:05.028986 140578044125248 machines.py:138] Terminating machine group. This may take up to a few minutes.
I1012 17:21:16.253354 140578044125248 machines.py:156] Machine group of 1 machines successfully terminated in 1.19 mins.

```

### CODE REVIEW

File: data_generation/estimate_costs.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                                 machine_group=machine_group)
+
+        # every 10 seconds, test if the simulation is done
+        task_status = task.get_status()
+        while task_status in ["submitted", "started"]:
+            time.sleep(10)
+            task_status = task.get_status()
+        # Compute the cost of the simulation
+        price_per_hour = machine_group.estimate_cloud_cost()
+        duration = task.get_execution_time()
```
Comment (IvanPombo) [inline data_generation/estimate_costs.py:78]: There is already an issue open with this request. Comment there to bring this up again :)

**Review by IvanPombo — APPROVED** [review]
LGTM!

File: data_generation/estimate_costs.py (hunk 1 above)
Comment (AugustoPeres) [reply data_generation/estimate_costs.py:78]: Can you be more specific on the issue? Is it this one: https://github.com/inductiva/inductiva/issues/385

Comment (IvanPombo) [reply data_generation/estimate_costs.py:78]: Indeed!

**Comment by sarmento** [comment]
I have only a few superficial comments for now.
I know I will be a bit annoying, but I see a lot of *crumbs* in the messaging to the user.
Here are a few things I would like to see changed (some may be out of the scope of this PR, so I am calling other eyes: @begaiym-k @luispcunha @cristianacarpinteiro @david451lima ).

```
I1012 17:18:27.370823 140578044125248 machines.py:63] Registering machine group configuration with the API.
```
why "with the API"? What would be the alternative?

``
I1012 17:18:28.359893 140578044125248 machines.py:67] Registered machine group:
I1012 17:18:28.359977 140578044125248 machines.py:68]  > ID: bd36b6a1-617f-45e9-ab5c-6ad4f32587b3
I1012 17:18:28.360085 140578044125248 machines.py:69]  > Name: api-bd36b6a1-617f-45e9-ab5c-6ad4f32587b3
```

Why do we hav some comments starting with ">"?
And why are we repeating this information below? (or here, if below is where it should be printed)?

```
I1012 17:18:28.360340 140578044125248 machines.py:114] Starting a machine group. This may take up to a few minutes.
I1012 17:18:28.360370 140578044125248 machines.py:116] Note that stopping this local process will not interrupt the creation of the machine group. Please wait...
I1012 17:18:53.193711 140578044125248 machines.py:124] Machine group successfully started in 0.41 mins.
```
what is "0.41 mins"?
Is this ~ 24 seconds (41% of 60 seconds)?
Is this 0 minutes and 41 seconds?

Why are we reporting time this way? It's not understandable... 

```
I1012 17:18:53.193801 140578044125248 machines.py:211] Name: api-bd36b6a1-617f-45e9-ab5c-6ad4f32587b3
I1012 17:18:53.193840 140578044125248 machines.py:212] Machine type: c2-standard-16
I1012 17:18:53.193861 140578044125248 machines.py:213] Number of machines: 1
I1012 17:18:53.193880 140578044125248 machines.py:214] Spot: False
I1012 17:18:53.193897 140578044125248 machines.py:215] Disk size: 40 GB
I1012 17:18:53.726830 140578044125248 machines.py:216] Estimated cost per hour: 0.919 $/h
```
Observe that here, we are reporting costs with up to 3 decimal cases and we have clearly understandable units. This is great. This should set the standard, right? Now look below...

```
I1012 17:18:53.827354 140578044125248 methods.py:284] Task ID: 1697127533798800741
I1012 17:18:53.827453 140578044125248 methods.py:81] Creating input archive...
I1012 17:18:53.832679 140578044125248 methods.py:88] Input archive created.
I1012 17:18:53.832753 140578044125248 methods.py:89] Input archive size: 27.12 KiB
```
Are we reporting in https://simple.wikipedia.org/wiki/Kibibyte ?
I am not sure this is the most thing... @hpenedones ? I think we need to have a comment about this somewhere, maybe in the FAQ?
```

I1012 17:18:53.832781 140578044125248 methods.py:92] Uploading input...
I1012 17:18:54.297517 140578044125248 methods.py:104] Input archive uploaded.
```
Since we hinted that this would take time with "..." then I think we should have said something like:

Input archive uploaded in X seconds.
```

I1012 17:18:54.297678 140578044125248 methods.py:296] Task request submitted.
I1012 17:20:04.983391 140578044125248 machines.py:189] Estimated cloud cost per hour: 0.919 $/h
```
ok! 3 decimal cases.
```
I1012 17:20:05.028886 140578044125248 estimate_costs.py:80] Simulation duration: 44.684656 seconds
```

:)

why 7 decimal cases? Do we even need more that 1 or 2 for this?

```
I1012 17:20:05.028962 140578044125248 estimate_costs.py:82] Estimated cost per task: 0.011406999684444443
```
1.140 cents?
We need to find a way of reporting quantities that makes sense. This number is unreadable.

```
I1012 17:20:05.028986 140578044125248 machines.py:138] Terminating machine group. This may take up to a few minutes.
I1012 17:21:16.253354 140578044125248 machines.py:156] Machine group of 1 machines successfully terminated in 1.19 mins.
```
~72 seconds or 79 seconds?


None of this is severe, but it accumulates and makes everything harder to understand.
Please don't let this go through.

**Comment by AugustoPeres** [comment]
> why 7 decimal cases? Do we even need more that 1 or 2 for this?

This last part is actually one of my logs. Will change it now

**Pull request merged**